### PR TITLE
Update trajectory-generator and move base trajectories out of sot-torque-control to separate package

### PR DESCRIPTION
**Pull request by proyan**
* Adds playSpline to load splines from a file
* Adds required dependency to [stack-of-tasks/parametric-curves](https://github.com/stack-of-tasks/parametric-curves) package
* Moves nd-TrajectoryGenerator base trajectories out of sot-torque-control to parametric-curves
* Everything except Text-File Trajectory Generator tested on simulation.
TODO:
Currently, it disables constant-acceleration and triangle-trajectory (still to be impemented in parametric-curves)

### CODE REVIEW

**Comment by proyan** [comment]
* constant-acceleration trajectory tested and activated in last commit

**Review by andreadelprete — CHANGES_REQUESTED** [review]
Everything looks fine to me except the fact that you have a single variable m_infiniteTime, whereas I think you should have one of these variables for each axis.

File: scripts/python_commands.txt [hunk 1]
```diff
@@ -29,14 +29,21 @@ create_topic(robot.ros, robot.device.forceRLEG,       'forceRLEG');
...
 create_topic(robot.ros, robot.device.accelerometer,       'acc');
+#create_topic(robot.ros, robot.torque_ctrl.dq_motor,       'dq_motor');
+#create_topic(robot.ros, robot.torque_ctrl.jointsAccelerationsDesired,       'ddq_des');
 
+#robot.torque_ctrl.velLeakingRate.value = 30*(0.,)
+#robot.torque_ctrl.reset_ddq_integral();
 ent.ctrl_manager.setCtrlMode('all','torque')
 robot.base_estimator.set_imu_weight(0.0);
 
+robot.com_traj_gen.playSpline("/local/rbudhira/git/proyan/sot-torque-control/scripts/spline1.out")
```
[the remark this answers is not in the document]
Comment (proyan) [reply scripts/python_commands.txt]: Done.

File: scripts/spline_trajectory_maker.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    A[i+2*len(x),2:] = np.array([n*(n-1)*t**(n-2) for n in xrange(2,deg_y+1)]);
+    A[i+2*len(x),0:2] = 0.0;
+
+  B = np.concatenate((y, dy, ddy), axis=1).transpose()
+  x, residual, _, _ = lstsq(A, B)
+  assert(residual <=eps)
+  return x
+
+
+CONTACT_SEQUENCE_WHOLEBODY_FILE="/local/rbudhira/git/jcarpent/reachability/wb_contact_sequence/contact_sequence_trajectory.xml"
```
[the remark this answers is not in the document]
Comment (proyan) [reply scripts/spline_trajectory_maker.py]: I made them relative to the package.

File: include/sot/torque_control/nd-trajectory-generator.hh [hunk 3]
```diff
@@ -146,26 +157,30 @@ namespace dynamicgraph {
...
-          JTG_TRIANGLE,
+          //JTG_TRIANGLE,
           JTG_CONST_ACC,
-          JTG_TEXT_FILE
+          JTG_TEXT_FILE,
+          JTG_SPLINE
         };
 
         bool              m_initSucceeded;    /// true if the entity has been successfully initialized
+        bool              m_infiniteTime;    /// true if trajectory is to go on forever
```
[the remark this answers is not in the document]
Comment (proyan) [reply include/sot/torque_control/nd-trajectory-generator.hh]: I think a better way might be setting infinite as the tmax for the trajectories. I'll do that.

**Pull request merged**